src/QMCDrivers/MCCoords.hpp

```diff
@@ -0,0 +1,129 @@
+//////////////////////////////////////////////////////////////////////////////////////
+// This file is distributed under the University of Illinois/NCSA Open Source License.
+// See LICENSE file in top directory for details.
+//
+// Copyright (c) 2022 developers.
+//
+// File developed by: Peter Doak, doakpw@ornl.gov, Oak Ridge National Laboratory
+//
+// File created by: Peter Doak, doakpw@ornl.gov, Oak Ridge National Laboratory
+//////////////////////////////////////////////////////////////////////////////////////
+
+#ifndef QMCPLUSPLUS_MCCOORDS_HPP
+#define QMCPLUSPLUS_MCCOORDS_HPP
+
+#include "Configuration.h"
+#include "type_traits/complex_help.hpp"
+#include "ParticleBase/RandomSeqGenerator.h"
+
+#include <vector>
+
+namespace qmcplusplus
+{
+
+enum class MCCoordsTypes
+{
+  RS,
+  RSSPINS
+};
+
+template<MCCoordsTypes MCT = MCCoordsTypes::RS>
+struct MCCoords
+{
+  static constexpr MCCoordsTypes mct = MCT;
+  // This cleans up some other code.
+  void resize(const std::size_t size);
+  std::vector<QMCTraits::PosType> rs;
+};
+
+template<>
+struct MCCoords<MCCoordsTypes::RSSPINS>
+{
+  static constexpr MCCoordsTypes mct = MCCoordsTypes::RSSPINS;
+  // This cleans up some other code.
+  void resize(const std::size_t size)
+  {
+    rs.resize(size);
+    spins.resize(size);
+  }
+  std::vector<QMCTraits::PosType> rs;
+  std::vector<std::complex<QMCTraits::RealType>> spins;
```

Comment:
spins is real not complex. So please update this and remove added complex related code.

Reply:
just for context, the spin variable in our formalism is real as Ye says. but the spin enters the wavefunction via spinors that we define as phi(r)_up exp(i * s) + phi(r)_dn exp(-i * s). So the coordinate is real, but it causes the wavefunction to be complex...which may have been the source of the confusion.

Reply:
Ok I think I saw it defined as complex in an early version of #3762 but I see that now if we drill down, scalar actually means real. 

The issue with Complex and the sequences should get addressed. The addition needed for real deltas creates an API that produces incorrect random complex numbers.